app/dao/notifications_dao.py

```diff
@@ -421,3 +424,32 @@ def get_total_sent_notifications_in_date_range(start_date, end_date, notificatio
     ).scalar()
 
     return result or 0
+
+
+def get_count_of_slow_delivery_sms_notifications_for_provider(
```

Comment:
So I wanted the function to not only tell me whether there was slow delivery, but also how many culprit notifications there were. Although, I do think a bool func is cleaner along with the separation. Will update.